File: Java Advanced/SetsMapsAdvanced/src/Phonebook.java
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Phonebook {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Map<String, String> phonebook = new LinkedHashMap<>();

        while (true){
            String input = scanner.nextLine();
            if(input.equals("search")){break;}
            String name = input.split("-")[0];
            String phoneNumber = input.split("-")[1];
            phonebook.put(name, phoneNumber);
        }

        while (true){
            String input = scanner.nextLine();
            if(input.equals("stop")){break;}

            if (phonebook.containsKey(input)) {
                System.out.printf("%s -> %s%n", input, phonebook.get(input));
            } else {
                System.out.printf("Contact %s does not exist.%n", input);
            }
        }
    }
}

File: JS Front-End/Objects and Classes/PhoneBook.js
function printPhonebook(input){
    let phoneBook = {};
    for(let token of input){
        let name = token.split(' ')[0];
        let number = token.split(' ')[1];
        phoneBook[name] = number;
    }

    for(let key in phoneBook){
        console.log(`${key} -> ${phoneBook[key]}`);
    }
}

printPhonebook(['George 0552554',
'Peter 087587',
'George 0453112',
'Bill 0845344']);